templates/flutter/lib/src/realtime_mixin.dart.twig

```diff
@@ -36,14 +41,16 @@ mixin RealtimeMixin {
       if (_lastUrl == uri.toString() && _websok?.closeCode == null) {
         return;
       }
+      _notifyDone = false;
```

Comment:
To prevent `_createSocket()` executing multiple times, what if we add a `bool _creatingSocket` and change this block to:


```diff
++  if (_creatingSocket) return;
++  _creatingSocket = true;
    final uri = _prepareUri();
    if (_websok == null) {
      _websok = await getWebSocket(uri);
      _lastUrl = uri.toString();
    } else {
      if (_lastUrl == uri.toString() && _websok?.closeCode == null) {
++      _creatingSocket = false;
        return;
      }
      _notifyDone = false;
      await _closeConnection();
      _lastUrl = uri.toString();
      _websok = await getWebSocket(uri);
      _notifyDone = true;
    }
    debugPrint('subscription: $_lastUrl');
```

and then add a finally at the end of the try in here;

```diff
    } catch (e) {
      if (e is AppwriteException) {
        rethrow;
      }
      if (e is WebSocketChannelException) {
        throw AppwriteException(e.message);
      }
      throw AppwriteException(e.toString());
--  }
++  } finally {
++    _creatingSocket = false;
++  }
```

Reply:
Hmm, seems like good idea.